Regex Extended / Group Specific Flags Fixes

wied03 opened the pull request:
* Construct Regexs in shared/new as objects so that we can put the escape sequence in quotes, otherwise Opal can't even compile the file, much less ignore it
* Wrap flags unsupported by Javascript Regex (extended)
* In Javascript, RE's cannot override flags within a group, the flags apply to the entire expression, so these are not supported

vais commented on core/regexp/shared/new_ascii_8bit.rb:
```diff
@@ -517,6 +537,6 @@ class RegexpSpecsSubclassTwo < Regexp; end
 
   it "sets the encoding to source String's encoding if the Regexp literal has the 'n' option and the source String is not ASCII only" do
-    Regexp.send(@method, /\xff/n).encoding.should == Encoding::ASCII_8BIT
+    Regexp.send(@method, Regexp.new("\xff", nil, 'n')).encoding.should == Encoding::ASCII_8BIT
```
This does not appear to be inter-changeable:
```ruby
Regexp.new("\xff", nil, 'n') == /\xff/n #=> false
```
Should be this instead:
```ruby
Regexp.new("\\xff", nil, 'n') == /\xff/n #=> true
```

vais commented on core/regexp/shared/new_ascii_8bit.rb:
```diff
@@ -80,7 +92,9 @@ class RegexpSpecsSubclassTwo < Regexp; end
     (r.options & Regexp::MULTILINE).should      == 0
-    (r.options & Regexp::EXTENDED).should       == 0
+    not_supported_on :opal do
+      (r.options & Regexp::EXTENDED).should       == 0
```
Clean up the white space before `==` to match the lines above?

vais commented on core/regexp/shared/new_ascii_8bit.rb:
```diff
@@ -37,36 +37,48 @@ class RegexpSpecsSubclassTwo < Regexp; end
-    (r.options & Regexp::EXTENDED).should_not   == 1
+    not_supported_on :opal do
+      r = Regexp.send(@method, 'Hi', Regexp::EXTENDED)
+      (r.options & Regexp::IGNORECASE).should     == 0
```
Clean up the white space before `==` to match the lines above?

vais commented on core/regexp/shared/new_ascii.rb:
```diff
@@ -424,6 +444,6 @@ class RegexpSpecsSubclassTwo < Regexp; end
 
   it "sets the encoding to source String's encoding if the Regexp literal has the 'n' option and the source String is not ASCII only" do
-    Regexp.send(@method, /\xff/n).encoding.should == Encoding::ASCII_8BIT
+    Regexp.send(@method, Regexp.new("\xff", nil, 'n')).encoding.should == Encoding::ASCII_8BIT
```
This does not appear to be inter-changeable:
```ruby
Regexp.new("\xff", nil, 'n') == /\xff/n #=> false
```
Should be this instead:
```ruby
Regexp.new("\\xff", nil, 'n') == /\xff/n #=> true
```

wied03 commented:
@vais - sorry for the sloppiness there. I updated the white space that was misaligned due to these changes and also fixed the quote I forgot to escape. I did test that against MRI but forgot to ensure just the regex was equivalent.

vais commented:
Thank you, @wied03 :sparkles:
